src/NUnitFramework/framework/TestContext.cs

```diff
@@ -326,6 +326,15 @@ public string ClassName
             }
 
             /// <summary>
+            /// Gets the name of the TestWorker that
+            /// is executing this test.
+            /// </summary>
+            public string WorkerId
+            {
+                get { return _test.WorkerId; }
```

Comment:
Obviously, you had to do the stuff with test, since you wanted to eventually write this line. But _test isn't really a test, it's an adapter class that holds only some items from the test on a read-only basis. If we want to expose details of how the test is being run to the user code somewhere else in the context.